tests/integration/test_kinesis.py

```diff
@@ -305,6 +311,105 @@ def test_record_lifecycle_data_integrity(
         for response_record in response_records:
             assert response_record.get("Data").decode("utf-8") in records_data
 
+    @pytest.mark.aws_validated
+    def test_subscribe_to_shard_timeout(
```

Comment:
There doesn't seem to be a limiting factor in the HTTP framework - I tested it on the full 5minutes locally and on AWS.
My best guess is that it's because the HTTP request does not actually stay open server-side... it only opens a response, once new records are found..